src/webapp/cilogon_ldap.py

```diff
@@ -42,3 +42,25 @@ def cilogon_id_map_to_ssh_keys(m):
         if 'sshPublicKey' in v['data']
     }
 
+
+def _entry2cinfo(entry):
+    ci = {'DNs': [entry['dn']]}
+    emails = entry['data'].get('mail')
+    if emails:
+        ci['PrimaryEmail'] = emails[0].lower()
+        if len(emails) >= 2:
+            ci['SecondaryEmail'] = emails[1].lower()
+    else:
```

Comment:
At least one email address is required; if you don't find one, put a warning in the log and skip the person.

Reply:
note that you need an email address to register with COManage in the first place so if this branch gets hit, something is wrong...

Reply:
Yes well, there happen to be two entries (both for Oksana Shadura) that do not have email addresses at the moment:

```
  "OSG1000041": {
    "data": {
      "cn": [
        "Oksana Shadura"
      ],
      "givenName": [
        "Oksana"
      ],
      "isMemberOf": [
        "CO:members:all",
        "CO:COU:OpenData Facility:members:all",
        "CO:members:active",
        "CO:COU:OpenData Facility:members:active",
        "opendata-facility-sponsors"
      ],
      "objectClass": [
        "person",
        "organizationalPerson",
        "inetOrgPerson",
        "eduMember",
        "voPerson"
      ],
      "sn": [
        "Shadura"
      ],
      "voPersonID": [
        "OSG1000041"
      ]
    },
    "dn": "voPersonID=OSG1000041,ou=people,o=OSG,o=CO,dc=cilogon,dc=org"
  },
```

```
  "OSG1000044": {
    "data": {
      "cn": [
        "Oksana Shadura"
      ],
      "givenName": [
        "Oksana"
      ],
      "isMemberOf": [
        "CO:members:all",
        "CO:COU:OpenData Facility:members:all",
        "CO:members:active",
        "CO:COU:OpenData Facility:members:active"
      ],
      "objectClass": [
        "person",
        "organizationalPerson",
        "inetOrgPerson",
        "eduMember",
        "voPerson"
      ],
      "sn": [
        "Shadura"
      ],
      "voPersonID": [
        "OSG1000044"
      ]
    },
    "dn": "voPersonID=OSG1000044,ou=people,o=OSG,o=CO,dc=cilogon,dc=org"
  }
```

Reply:
> At least one email address is required; if you don't find one, put a warning in the log and skip the person.

Sounds good!

Reply:
contacts with no email are omitted now